### PR TITLE
Add validation for variable kind - fix_env and fix_variable.

### LINKED ISSUE #2183: validate value for fix_env and fix_variable variables
for deployment variables in OCL which have kind - fix_env and fix_variable, we must always validate that the default value is provided by the ISV. 

- [ ] Add validation
- [ ] throw error  - MandatoryValueMissingForFixedVariables and return the variables which have values missing.

### PR DESCRIPTION
**Pull request by Alice1319**
fixes #2183 
updated:
image

Snipaste_2024-12-13_16-16-47

### CODE REVIEW

File: modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/utils/DeployVariableSchemaValidator.java [hunk 1]
```diff
@@ -46,6 +61,14 @@ public static void validateDeployVariable(List<DeployVariable> deployVariables)
                 varNameSet.add(deployVariable.getName());
             }
         }
+        if (!missingExampleValueVariables.isEmpty()) {
+            String errorMessage = "The deploy variable configuration for "
+                    + DeployVariableKind.FIX_ENV.toValue() + " or " +
+                    DeployVariableKind.FIX_VARIABLE.toValue()
+                    + " kind has an empty example.";
```
Comment (swaroopar) [inline modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/utils/DeployVariableSchemaValidator.java:68]: its better we have a more clear error message - 
Variable {variable_name} has type {variable_type} but a value for it is not provided.

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/exceptions/MandatoryValueMissingForFixedVariablesException.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+/**
+ * Exception thrown when a mandatory value missing for fixed variables.
+ */
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class MandatoryValueMissingForFixedVariablesException extends RuntimeException {
```
Comment (swaroopar) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/exceptions/MandatoryValueMissingForFixedVariablesException.java:16]: please also had an handler in RegistrationExceptionHandler

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Comment by WangLiNaruto** [comment]
LGTM

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/exceptions/MandatoryValueMissingForFixedVariablesException.java (hunk 2 above)
Comment (Alice1319) [reply modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/exceptions/MandatoryValueMissingForFixedVariablesException.java]: fixed.

File: modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/utils/DeployVariableSchemaValidator.java (hunk 1 above)
Comment (Alice1319) [reply modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/utils/DeployVariableSchemaValidator.java]: fixed.

Comment (Alice1319) [reply modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/utils/DeployVariableSchemaValidator.java]: image

**Comment by wuzhuanhong** [comment]
LGTM

**Comment by profoundwu** [comment]
LGTM

**Comment by baixinsui** [comment]
LGTM

**Comment by yuanxiaojun987** [comment]
LGTM

**Comment by luoping-12345** [comment]
LGTM

File: modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/utils/DeployVariableSchemaValidator.java [hunk 3]
```diff
@@ -46,6 +63,27 @@ public static void validateDeployVariable(List<DeployVariable> deployVariables)
                 varNameSet.add(deployVariable.getName());
             }
         }
+        if (!missingExampleValueVariables.isEmpty()) {
+            StringBuilder errorMessageBuilder =
+                    new StringBuilder("The deploy variable " + "configuration for ")
+                            .append(DeployVariableKind.FIX_ENV.toValue())
+                            .append(" or ")
+                            .append(DeployVariableKind.FIX_VARIABLE.toValue())
+                            .append(" kind has an empty example. Details: ");
```
Comment (swaroopar) [inline modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/utils/DeployVariableSchemaValidator.java:72]: Should be - empty default value.

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/utils/DeployVariableSchemaValidator.java (hunk 3 above)
Comment (Alice1319) [reply modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/utils/DeployVariableSchemaValidator.java]: updated.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**